akka-samples/akka-sample-cluster-scala/src/main/scala/sample/cluster/factorial/FactorialFrontend.scala

```diff
@@ -61,10 +61,18 @@ object FactorialFrontend {
     //#registerOnRemoved
     Cluster(system).registerOnMemberRemoved {
       // exit JVM when ActorSystem has been terminated
-      system.registerOnTermination(System.exit(-1))
+      system.registerOnTermination(System.exit(0))
       // in case ActorSystem shutdown takes longer than 10 seconds,
-      // exit the JVM forcefully anyway
-      system.scheduler.scheduleOnce(10.seconds)(System.exit(-1))(system.dispatcher)
+      // exit the JVM forcefully anyway.
+      // Use a Deadline since the scheduled task will also be run when
+      // the ActorSystem is terminated normally.
+      val shutdownDeadline = Deadline.now + 10.seconds
+      system.scheduler.scheduleOnce(shutdownDeadline.timeLeft + 10.millis) {
+        if (shutdownDeadline.isOverdue())
```

Comment:
But the JVM timer is not dangerous here since
 - it is only used for one task and only during shutdown
 - memory leak is not relevant since the JVM will be killed anyway

Another alternative is to spawn a deamon thread that is calling the Await for the system shutdown and aborts if necessary.